nomad/resource_job_test.go

```diff
@@ -208,6 +223,7 @@ func testResourceJob_checkExists(s *terraform.State) error {
 
 func testResourceJob_checkDestroy(jobID string) r.TestCheckFunc {
 	return func(*terraform.State) error {
+		time.Sleep(time.Second) // let things happen in background
```

Comment:
What things? What are we waiting for here? Why is this necessary?

Sometimes Sleeps are necessary, and it's always unfortunate when that happens, but if we document why, at least we can remove them in the future when they stop being necessary.

Reply:
This test is failing on the recent commit of the master branch of this repo for me.

I'm not sure, whether this test was written in such way itnentionally. I might have misunderstood the whole idea of this test.

But, anyway, my understanding is:
* in test we create a job with command `/bin/sleep 1`
* right after we create the job we check that the job is "dead"
* apparently, by the time we check job state, it is still not "dead", so test fails

So I introduced sleep before we check.

Perhaps, I'd better change job definition to `/bin/true`, so the test will pass (in most cases, explained below) without introducing delay before the check.

This solution will practically work, but is still conceptually wrong and hypothetically may fail on a very slow machine.

But, I don't see, how the previous version of the test have been working at all.

Let me know, what do you think.

Reply:
Appreciate the clarification, and thanks for fixing the test!

I'm uncomfortable with the Sleep, and think a better idea would probably be to introduce some polling to make things a bit more robust (and agree that `/bin/true` is probably a better test job to run), but it's a bit removed from the main problem this PR is addressing, so I'm ok just letting it pass for the moment and circling back around to fix it after we get this resolved.